### PR TITLE
Remove unused RCW/CCW code paths

### PR DESCRIPTION
**Pull request by AaronRobinsonMSFT**
/cc @jkoritzinsky @elinor-fung @jkotas

### CODE REVIEW

**Comment by AaronRobinsonMSFT** [comment]
@davmason, @noahfalk, @mikem8361 I updated DAC code to return "default" values since some queries are no longer valid.

File: src/coreclr/src/vm/comcallablewrapper.cpp [hunk 1]
```diff
@@ -906,31 +889,6 @@ SimpleComCallWrapper::~SimpleComCallWrapper()
...
-        THROWS;
-        GC_TRIGGERS;
-        MODE_ANY;
-        INJECT_FAULT(COMPlusThrowOM());
-        POSTCONDITION(CheckPointer(RETVAL));
-    }
-    CONTRACT_END;
-
-    NewHolder<SimpleComCallWrapper> pWrapper = new SimpleComCallWrapper;
-    memset (pWrapper, 0, sizeof(SimpleComCallWrapper));
```
Comment (AaronRobinsonMSFT) [inline src/coreclr/src/vm/comcallablewrapper.cpp:928]: The `SimpleComCallWrapper` constructor calls this to zero out itself.

File: src/coreclr/src/debug/daccess/request.cpp [hunk 2]
```diff
@@ -4343,13 +4343,6 @@ HRESULT ClrDataAccess::IsRCWDCOMProxy(CLRDATA_ADDRESS rcwAddr, BOOL* isDCOMProxy
     *isDCOMProxy = FALSE;
 
 #ifdef FEATURE_COMINTEROP
```
Comment (davmason) [inline src/coreclr/src/debug/daccess/request.cpp:4345]: Would it make more sense to remove the #ifdef?

**Review by davmason — APPROVED** [review]
I don't know enough about the COM stuff to comment on it, but from a DAC coding perspective I don't see any issues

File: src/coreclr/src/debug/daccess/request.cpp (hunk 2 above)
Comment (AaronRobinsonMSFT) [reply src/coreclr/src/debug/daccess/request.cpp:4345]: @davmason I can do that. I didn't change it simple to ensure the behavior of the API is the same when COM isn't supported. I assume SOS would indicate "command failed" now and if I change to `S_OK` then would say "Not a COM proxy", but COM isn't support anyways. Really is up to you, I can remove it and always return `E_NOTIMPL` or `S_OK`.

Comment (davmason) [reply src/coreclr/src/debug/daccess/request.cpp:4345]: For some reason I was thinking that FEATURE_COMINTEROP was always defined, but of course it isn't on linux/mac. So I think the way you have it makes sense.

File: src/coreclr/src/vm/cominterfacemarshaler.cpp [hunk 3]
```diff
@@ -456,6 +385,29 @@ VOID COMInterfaceMarshaler::InitializeExistingComObject(OBJECTREF *pComObj, IUnk
...
+    }
+    CONTRACTL_END;
+
+    // If the marshaller's type handle is null, compute what it should be.
+    if (m_typeHandle.IsNull())
+    {
+        // We no longer support the mapping from IUnknown to Type via
+        // IProvideClassinfo in CoreCLR., so if the type handle isn't
+        // set then fallback to the __ComObject type.
+        m_typeHandle = TypeHandle(g_pBaseCOMObject);
```
Comment (elinor-fung) [inline src/coreclr/src/vm/cominterfacemarshaler.cpp:407]: I'm missing what exactly the `IProvideClassInfo` support here meant and why it is fine to remove in this case. Can you elaborate / comment on what scenarios are or are not supported in relation to `IProvideClassInfo`?

File: src/coreclr/src/vm/comcallablewrapper.cpp [hunk 4]
```diff
@@ -3719,78 +3556,6 @@ void ComMethodTable::LayOutClassMethodTable()
...
-        THROWS;
-        GC_TRIGGERS;
-        MODE_ANY;
-        PRECONDITION(CheckPointer(pItfClass));
-    }
-    CONTRACTL_END;
-
-    if (!IsLayoutComplete())
-    {
-        if (!IsSigClassLoadChecked())
```
Comment (elinor-fung) [inline src/coreclr/src/vm/comcallablewrapper.cpp:3736]: Looks like `IsSigClassLoadChecked`, `SetSigClassLoadChecked`, `IsSigClassCannotLoad`, and `SetSigClassCannotLoad` can be removed too.

**Review by elinor-fung — APPROVED** [review]

File: src/coreclr/src/vm/cominterfacemarshaler.cpp (hunk 3 above)
Comment (AaronRobinsonMSFT) [reply src/coreclr/src/vm/cominterfacemarshaler.cpp:407]: The [`IProvideClassInfo`](https://docs.microsoft.com/windows/win32/api/ocidl/nn-ocidl-iprovideclassinfo) interface provided an `ITypeInfo` instance that we would then attempt to convert to a `MethodTable` instance. All the logic for mapping and storing `ITypeInfo` <=> `MethodTable` was removed with the `TlbExp` support. I will update the comment. For more context see the removed function `GetClassFromIProvideClassInfo` in this PR.

**Pull request merged**